src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/ApiTests/AuthorizationApiTests.cs

```diff
@@ -0,0 +1,187 @@
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "/tdd/endring-av-navn/api/authorization/parties/current?returnPartyObject=true")
+            {
+            };
+
+            AddPartyCookie(httpRequestMessage, 500003);
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(System.Net.HttpStatusCode.OK, response.StatusCode);
+            Party party = JsonConvert.DeserializeObject<Party>(responseContent);
+            Assert.Equal(500003, party.PartyId);
+        }
+
+        /// <summary>
+        /// Scenrio: User is not authorized for the party in cookie
+        /// </summary>
+        [Fact]
+        public async Task GetCurrentPartyWithPartyCookieForOther_ReturnSelfPartyID_OK()
+        {
+            string token = PrincipalUtil.GetToken(1);
+            HttpClient client = SetupUtil.GetTestClient(_factory, "tdd", "endring-av-navn");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "/tdd/endring-av-navn/api/authorization/parties/current")
+            {
+            };
+
+            AddPartyCookie(httpRequestMessage, 500003);
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(System.Net.HttpStatusCode.OK, response.StatusCode);
+            int partyId = JsonConvert.DeserializeObject<int>(responseContent);
+            Assert.Equal(1000, partyId);
+        }
+
+        /// <summary>
+        /// Scenrio: User is not authorized for the party in cookie
+        /// </summary>
+        [Fact]
+        public async Task GetCurrentPartyWithPartyCookieForOther_ReturnSelfParty_OK()
+        {
+            string token = PrincipalUtil.GetToken(1);
+            HttpClient client = SetupUtil.GetTestClient(_factory, "tdd", "endring-av-navn");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "/tdd/endring-av-navn/api/authorization/parties/current?returnPartyObject=true")
+            {
+            };
+
+            AddPartyCookie(httpRequestMessage, 500003);
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);
+            string responseContent = await response.Content.ReadAsStringAsync();
+
+            Assert.Equal(System.Net.HttpStatusCode.OK, response.StatusCode);
+            Party party = JsonConvert.DeserializeObject<Party>(responseContent);
+            Assert.Equal(1000, party.PartyId);
```

Comment:
Should this test also assert that the cookie was updated with the correct partyId?

Reply:
added